py/miniconf-mqtt/miniconf/miniconf.py

```diff
@@ -44,41 +42,22 @@ def __init__(self, client, prefix):
         self.client = client
         self.prefix = prefix
         self._inflight = {}
-        self.client.on_message = self._on_message
-        self.client.on_subscribe = self._on_subscribe
         self.response_topic = f"{prefix}/response"
-        self._pending_subscriptions = {
-            self.client.subscribe(
-                self.response_topic
-            ): asyncio.get_running_loop().create_future(),
-        }
-
-    async def subscriptions_complete(self):
-        """Wait for all pending subscriptions to complete."""
-        return await asyncio.gather(*self._pending_subscriptions.values())
-
-    def _on_subscribe(self, _client, mid, _qos, _props):
-        LOGGER.info("Handling subscription for %s", mid)
-        if mid not in self._pending_subscriptions:
-            LOGGER.warning("MID: %s, unexpected subscription", mid)
-            return
+        self.subscribed = False
 
-        self._pending_subscriptions[mid].set_result(True)
-        del self._pending_subscriptions[mid]
 
-    def _on_message(self, _client, topic, payload, _qos, properties):
-        """Callback function for when messages are received over MQTT.
+    def process_message(self, message: Message):
+        """ Process a received MQTT message. """
+        try:
+            properties = message.properties.json()
+        except AttributeError:
+            properties = {}
+
+        LOGGER.debug(f"Received {message.topic}: {message.payload} [{properties}]")
```

Comment:
IIRC one reason behind the f-string lint for logging is that it is expensive compared to lazy formatting even if the message ends up being discarded.
I'd like to keep the high-traffic situation in mind.

Reply:
Fixed in f4de72322c26cebfb5b53bd81172b1a8138c0f81